File: js/templates/reply-automation/src/classify.ts
/**
 * A tiny, dependency-free reply classifier. Swap in your own model or call
 * Instantly's `testAiReplyLabelLeadLabels` for AI labels (rate-limited: 500/30d).
 *
 * `interestValue` maps to a lead's `lt_interest_status` / `interest_value`:
 *   1 interested · 2 meeting booked · 0 neutral · -1 not interested · -2 wrong person
 */
export interface Classification {
  label: "interested" | "meeting" | "not_interested" | "wrong_person" | "out_of_office" | "neutral"
  interestValue: number
}

const RULES: Array<{ label: Classification["label"]; interestValue: number; patterns: RegExp[] }> = [
  { label: "not_interested", interestValue: -1, patterns: [/\bunsubscribe\b/i, /not interested/i, /\bstop\b/i, /remove me/i, /no thanks/i] },
  { label: "wrong_person", interestValue: -2, patterns: [/wrong person/i, /not the right/i, /no longer with/i, /reach out to/i] },
  { label: "out_of_office", interestValue: 0, patterns: [/out of office/i, /\bOOO\b/, /on leave/i, /on vacation/i, /annual leave/i] },
  { label: "meeting", interestValue: 2, patterns: [/book (a )?(call|meeting|time)/i, /calendar/i, /calendly/i, /what times?/i, /let'?s meet/i] },
  {
    label: "interested",
    interestValue: 1,
    patterns: [
      /interested/i, /tell me more/i, /sounds (good|great|interesting)/i, /learn more/i,
      /pricing/i, /how much/i, /send me (the )?(info|details|more|deck)/i, /yes[, ]*please/i,
      /sign me up/i, /count me in/i, /\bkeen\b/i, /let'?s (do it|chat|talk)/i,
    ],
  },
]

export function classify(text: string): Classification {
  const body = text ?? ""
  for (const rule of RULES) {
    if (rule.patterns.some((p) => p.test(body))) {
      return { label: rule.label, interestValue: rule.interestValue }
    }
  }
  return { label: "neutral", interestValue: 0 }
}

File: python/templates/reply-automation/app/classify.py
"""A tiny, dependency-free reply classifier. Swap in your own model or call
Instantly's test_ai_reply_label_lead_labels for AI labels (rate-limited: 500/30d).

interest_value maps to a lead's lt_interest_status / interest_value:
    1 interested · 2 meeting booked · 0 neutral · -1 not interested · -2 wrong person
"""
import re
from dataclasses import dataclass
from typing import List, Tuple


@dataclass
class Classification:
    label: str
    interest_value: int


_RULES: List[Tuple[str, int, List[str]]] = [
    ("not_interested", -1, [r"\bunsubscribe\b", r"not interested", r"\bstop\b", r"remove me", r"no thanks"]),
    ("wrong_person", -2, [r"wrong person", r"not the right", r"no longer with", r"reach out to"]),
    ("out_of_office", 0, [r"out of office", r"\bOOO\b", r"on leave", r"on vacation", r"annual leave"]),
    ("meeting", 2, [r"book (a )?(call|meeting|time)", r"calendar", r"calendly", r"what times?", r"let'?s meet"]),
    ("interested", 1, [
        r"interested", r"tell me more", r"sounds (good|great|interesting)", r"learn more",
        r"pricing", r"how much", r"send me (the )?(info|details|more|deck)", r"yes[, ]*please",
        r"sign me up", r"count me in", r"\bkeen\b", r"let'?s (do it|chat|talk)",
    ]),
]


def classify(text: str) -> Classification:
    body = text or ""
    for label, value, patterns in _RULES:
        if any(re.search(p, body, re.IGNORECASE) for p in patterns):
            return Classification(label=label, interest_value=value)
    return Classification(label="neutral", interest_value=0)
